### PR TITLE
dedup idna code: JSPs call unicodetools

### PR DESCRIPTION
**Pull request by markusicu**
Deduplicate Idna* code. Make the JSP code call unicodetools idna classes.
The `.idna` classes look more modern than the `.jsp` classes, so I just deleted the latter and adjusted calling code.
This includes class Uts46 after careful diffing, and checking with Mark.
I am also moving TestIdna.java and TestUts46.java to the unicodetools, near the code that they are testing.

### CODE REVIEW

**Review by srl295 — APPROVED** [review]
LGTM and yes TestIdna ought to move

**Review by macchiati — APPROVED** [review]
LGTM. BTW, adding 'final' is a fine thing to do, but nicer in a second PR since it makes it harder to separate out the substantive changes.

**Comment by markusicu** [comment]
> BTW, adding 'final' is a fine thing to do, but nicer in a second PR since it makes it harder to separate out the substantive changes.

The changes in Uts46.java are copied from the unicodetools version to the JSP version.
I made those changes in order to make the two versions easier to compare.
Based on your email reply, I will try to delete the JSP version altogether.
But first I need to look at the build breakages...

**Comment by macchiati** [comment]
Ah, makes sense.

**Comment by markusicu** [comment]
@macchiati I ran GenerateIdna and GenerateIdnaTest and compared the output with what I got before this PR (which is what's checked into the repo at .../idna/dev/).
There is a difference in reported errors on two lines of the test data file.
Using the .idna code we no longer see the P1 and V6 errors reported on these lines by the .jsp code.
[P1](https://www.unicode.org/reports/tr46/#Processing) is for a disallowed code point. The input includes U+31816 which is new in Unicode 15, so this looks like a bug fix.
[V6](https://www.unicode.org/reports/tr46/#Validity_Criteria) is similar:
```
            // For Transitional Processing, each value must be valid.
            // For Nontransitional Processing, each value must be either valid
            // or deviation.
```
So that could be due to the same new-in-15 code point, meaning this is good?

```patch
@@ -2051,8 +2051,8 @@
 ߢσ‍𝟳。蔑򛖢; ߢσ‍7.蔑򛖢; [B2, C2, P1, V6]; xn--7-zmb872aez5a.xn--wy1ao4929b; ; xn--7-zmb872a.xn--wy1ao4929b; [B2, P1, V6] # ߢσ7.蔑
 𐹰.؀; ; [B1, P1, V6]; xn--oo0d.xn--ifb; ; ;  # 𐹰.
 xn--oo0d.xn--ifb; 𐹰.؀; [B1, V6]; xn--oo0d.xn--ifb; ; ;  # 𐹰.
--ࢨ.𱠖; ; [B1, P1, V3, V6]; xn----mod.xn--5o9n; ; ;  # -ࢨ.𱠖
-xn----mod.xn--5o9n; -ࢨ.𱠖; [B1, V3, V6]; xn----mod.xn--5o9n; ; ;  # -ࢨ.𱠖
+-ࢨ.𱠖; ; [B1, V3]; xn----mod.xn--5o9n; ; ;  # -ࢨ.𱠖
+xn----mod.xn--5o9n; -ࢨ.𱠖; [B1, V3]; xn----mod.xn--5o9n; ; ;  # -ࢨ.𱠖
 ≯𞱸󠇀。誆⒈; ≯𞱸.誆⒈; [B1, P1, V6]; xn--hdh7151p.xn--tsh1248a; ; ;  # ≯𞱸.誆⒈
 ≯𞱸󠇀。誆⒈; ≯𞱸.誆⒈; [B1, P1, V6]; xn--hdh7151p.xn--tsh1248a; ; ;  # ≯𞱸.誆⒈
 ≯𞱸󠇀。誆1.; ≯𞱸.誆1.; [B1, P1, V6]; xn--hdh7151p.xn--1-dy1d.; ; ;  # ≯𞱸.誆1.
```

**Comment by markusicu** [comment]
I think I am done here. PTAL

**Comment by markusicu** [comment]
Could someone please re-approve this PR? In addition to the listed reviewers, maybe @nedley?

**Review by macchiati — APPROVED** [review]

**Comment by markusicu** [comment]
Thanks @macchiati !

File: .github/workflows/build-jsp.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     - name: Package JSPs
       run: >
-        mvn -s .github/workflows/mvn-settings.xml -B -pl UnicodeJsps install package
+        mvn -s .github/workflows/mvn-settings.xml -B -pl UnicodeJsps install package -DCLDR_DIR=${GITHUB_WORKSPACE}/cldr
```
Comment (srl295) [inline .github/workflows/build-jsp.yml:57]: Tests must be run here. Good catch.

**Review by srl295 — COMMENTED** [review]
LGTM

File: .github/workflows/build-jsp.yml (hunk 1 above)
Comment (markusicu) [reply .github/workflows/build-jsp.yml:57]: Tests were run here before, but across the several commits sometimes I needed the CLDR path and other times I didn't. :shrug:

**Pull request merged**